Dialog/balloon.gd

```diff
@@ -0,0 +1,176 @@
+# behavior added in to fit the needs for the game!
+
+## The action to use for advancing the dialogue
+@export var next_action: StringName = &"ui_accept"
+
+## The action to use to skip typing the dialogue
+@export var skip_action: StringName = &"ui_cancel"
+
+@onready var balloon: Control = %Balloon
+@onready var character_label: RichTextLabel = %CharacterLabel
+@onready var dialogue_label: DialogueLabel = %DialogueLabel
+@onready var responses_menu: DialogueResponsesMenu = %ResponsesMenu
+@onready var portrait: TextureRect = %Balloon/Panel/HBoxContainer/Control/Portrait
+
+# Custom variables for the flavor doodles/text on the dialog that can be customizable in the dialog script
+@onready var flavor_image_1: TextureRect = %Balloon/Panel/FlavorNode/FlavorImage1
+@onready var flavor_image_2: TextureRect = %Balloon/Panel/FlavorNode/FlavorImage2
+@onready var flavor_image_3: TextureRect = %Balloon/Panel/FlavorNode/FlavorImage3
+
+@onready var flavor_text_1: RichTextLabel = %Balloon/Panel/FlavorNode/FlavorText1
+@onready var flavor_text_2: RichTextLabel = %Balloon/Panel/FlavorNode/FlavorText2
+@onready var flavor_text_3: RichTextLabel = %Balloon/Panel/FlavorNode/FlavorText3
+
+## The dialogue resource
+var resource: DialogueResource
+
+## Temporary game states
+var temporary_game_states: Array = []
+
+## See if we are waiting for the player
+var is_waiting_for_input: bool = false
+
+## See if we are running a long mutation and should hide the balloon
+var will_hide_balloon: bool = false
+
+## The current line
+var dialogue_line: DialogueLine:
+	set(next_dialogue_line):
+		is_waiting_for_input = false
+		balloon.focus_mode = Control.FOCUS_ALL
+		balloon.grab_focus()
+
+		# The dialogue has finished so close the balloon
+		if not next_dialogue_line:
+			queue_free()
+			return
+
+		# If the node isn't ready yet then none of the labels will be ready yet either
+		if not is_node_ready():
+			await ready
+
+		dialogue_line = next_dialogue_line
+
+		character_label.visible = not dialogue_line.character.is_empty()
+		character_label.text = tr(dialogue_line.character, "dialogue")
+
+		dialogue_label.hide()
+		dialogue_label.dialogue_line = dialogue_line
+
+		responses_menu.hide()
+		responses_menu.set_responses(dialogue_line.responses)
+
+		# Show our balloon
+		balloon.show()
+		will_hide_balloon = false
+
+		dialogue_label.show()
+		if not dialogue_line.text.is_empty():
+			dialogue_label.type_out()
+			await dialogue_label.finished_typing
+
+		# Wait for input
+		if dialogue_line.responses.size() > 0:
+			balloon.focus_mode = Control.FOCUS_NONE
+			responses_menu.show()
+		elif dialogue_line.time != "":
+			var time: float = dialogue_line.text.length() * 0.02 if dialogue_line.time == "auto" else dialogue_line.time.to_float()
+			await get_tree().create_timer(time).timeout
+			next(dialogue_line.next_id)
+		else:
+			is_waiting_for_input = true
+			balloon.focus_mode = Control.FOCUS_ALL
+			balloon.grab_focus()
```

Comment:
Did you find this yourself or copied from an example ? I'm curious

Reply:
![image](https://github.com/Saplings-Projects/1M_sub/assets/5923102/ca9a9c87-c512-4cdf-932e-cb341ba534bb)

With the addition of the add-on, there's a way to make a copy of the dialogue balloon which generates this balloon.gd file.

Reply:
Humu, interesting